glom/core.py

```diff
@@ -2267,7 +2270,7 @@ def glom(target, spec, **kwargs):
         except skip_exc:
             if default is _MISSING:
                 raise
-            ret = default
+            ret = default  # should this also be arg_val'd?
```

Comment:
if we evaluate early, how bad would it be for perf? and if we evaluate it late (i.e., at this line), and it fails, that seems like it would be bad for reliability?

Reply:
yea, I'm inclined now to back off a little bit and leave things lay

Reply:
it was just a thought experiment -- if arg_val makes sense for all these internal spec defaults, what about the top level default?

I think you're right about reliability though, and also I think the entry point is a special case -- the entire glom "program" is meant to be contained in the spec, and I think that's an important top-level invariant